packages/@aws-cdk/aws-lambda/lib/lambda-insights.ts

```diff
@@ -65,6 +81,13 @@ export abstract class LambdaInsightsVersion {
    * The arn of the Lambda Insights extension
    */
   public readonly layerVersionArn: string = '';
+
+  /**
+   * Returns the arn of the Lambda Insights extension based on the
+   * Lambda architecture
+   * @internal
+   */
+  public abstract _bind(architecture?: Architecture): string;
```

Comment:
To be safe, should we add a couple more arguments here to make this easy to extend in the future? Typically we'll do something like:

```ts
public abstract bind(scope: Construct, function: IFunction): InsightBindProps;
```

* scope -> in case the integration needs to add constructs
* function -> to give access to function details, function could expose `function.architecture` 
* return props -> in case we need to add more props in the future

Other than that, I'm down with this.

Reply:
updated